opencog/openpsi/utilities.scm

```diff
@@ -326,13 +326,13 @@
 		result
 		; else check if entity is an evaluation or predicate or schema
 		(let ((type (cog-type entity)))
-			(if (or (equal? type 'PredicateNode)
-				   (equal? type 'GroundedPredicateNode)
+			(if (or (equal? type 'GroundedPredicateNode)
 				   (equal? type 'DefinedPredicateNode))
 				(set! result (cog-evaluate! (Evaluation entity (List)))))
-			(if (or (equal? type 'SchemaNode)
-            		(equal? type 'GroundedSchemaNode)
-            	    (equal? type 'DefinedSchemaeNode))
+			(if (equal? type 'PredicateNode)
+				(set! result (cog-tv entity)))
```

Comment:
In the case of a 0-ary predicate there is no meaningful difference between
s1 and s2 where

P <s1>

EvaluationLink <s3>
      P


... is there?  they should just be the same number...

The semantics of

P <s1>

is the average of P(x) over all arguments x, and there are no arguments...



On Mon, Aug 1, 2016 at 11:23 AM, Eddie Monroe <notifications@github.com>
wrote:

> In opencog/openpsi/utilities.scm
> <https://github.com/opencog/opencog/pull/2334#discussion_r73007889>:
>
> >  				   (equal? type 'DefinedPredicateNode))
> >  				(set! result (cog-evaluate! (Evaluation entity (List)))))
> > -			(if (or (equal? type 'SchemaNode)
> > -            		(equal? type 'GroundedSchemaNode)
> > -            	    (equal? type 'DefinedSchemaeNode))
> > +			(if (equal? type 'PredicateNode)
> > +				(set! result (cog-tv entity)))
>
> @bgoertzel <https://github.com/bgoertzel> What about the issue of (a)
> using EvaluationLink for assigning values to predicates with no arguments
> (for consistency with predicates that do have arguments) vs. (b) using the
> PredicateNode TV (rather than EvaluationLink) to assign value to predicates
> that don't have arguments?
>
> It seems to me there should be an OpenCog standard here. For example, if
> openpsi wants to know the value of a 0-ary predicate that is defined by a
> different component of the system (e.g., a PAU like voice-width), where
> should openpsi look to get it's value from, the PredicateNode TV or a
> wrapped EvaluationLink TV?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opencog/opencog/pull/2334/files/255ac40a46ca72d1fe5b4f88e3959a99a2c6658d#r73007889>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFolXDhI7JsJ3oQCqw-OX8ozgHeiKITiks5qbh2bgaJpZM4JYxpE>
> .
>



-- 
Ben Goertzel, PhD
http://goertzel.org

Super-benevolent super-intelligence is the thought the Global Brain is
currently struggling to form...